src/core/link-to-dfn.js

```diff
@@ -121,30 +139,15 @@ function collectDfns(title) {
       }
     }
     result.set(dfnFor, dfn);
-    assignDfnId(dfn, title);
+    addId(dfn, "dfn", title);
   });
   return { result, duplicates };
 }
 
-/**
- * @param {HTMLElement} dfn
- * @param {string} title
- */
-function assignDfnId(dfn, title) {
-  if (!dfn.id) {
-    const { dfnFor } = dfn.dataset;
-    if (dfn.dataset.idl) {
-      addId(dfn, "dom", (dfnFor ? `${dfnFor}-` : "") + title);
```

Comment:
I assume this line never ran, probably because `decorateDfn()` ran first, right?

Reply:
correct. At least, I was never able to trigger it.